### PR TITLE
LPS-167450 - DDL records is published to live when disabled Publish Displayed Content by Default

### PR DESCRIPTION
**Pull request by DaniSzimko**
[LPS-167450](https://liferay.atlassian.net/browse/LPS-167450) - modified the DDLDisplay portlet to fetch the records based on groupId and recordSetKey instead of recordSetId

Hello team,

I would like to request a review of this PR, as it changes the behavior of the DDL Display Portlet. I have made modifications to the **DDLDisplayExportImportPortletPreferencesProcessor** class to align its behavior with that of the **JournalContentExportImportPortletPreferencesProcessor**. Additionally, I've included the **groupId** and **recordSetKey** in the portlet preferences within the **DDLDisplayConfigurationAction** class.

The issue with the **DDLDisplayPortlet** was that it only worked with the **recordSetId**, without considering the **groupId**. Consequently, it couldn't differentiate between staging and live **recordSets**. When we published the page containing the **DDLDisplayPortlet** without the **recordSet**, the display portlet still pointed to the single existing **recordSet**, even though it should have been exclusive to the staging site. Now you can publish the **recordSet** and the **records** individually.

Update :
**Expected** **Behavior**: The portlets are present (you can see the divs in the inspector) on Live but not the contents from them.
**Actual** **Behavior**: The web content doesn't present but the DDL record does.

### CODE REVIEW

File: modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/exportimport/portlet/preferences/processor/DDLDisplayExportImportPortletPreferencesProcessor.java [hunk 1]
```diff
@@ -161,15 +217,75 @@ public PortletPreferences processImportPortletPreferences(
...
+				importedRecordSetGroup.isStagedPortlet(
+					DDLPortletKeys.DYNAMIC_DATA_LISTS)) {
+
+				Map<Long, Long> recordSetIds =
+					(Map<Long, Long>)portletDataContext.getNewPrimaryKeysMap(
+						DDLRecordSet.class);
+
+				long recordSetId = GetterUtil.getLong(importedRecordSetId);
+
+				recordSetId = MapUtil.getLong(recordSetIds, recordSetId, 0);
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-lists/dynamic-data-lists-web/src/main/java/com/liferay/dynamic/data/lists/web/internal/exportimport/portlet/preferences/processor/DDLDisplayExportImportPortletPreferencesProcessor.java:273]: @DaniSzimko please inline:
```
long recordSetId = MapUtil.getLong(recordSetIds, GetterUtil.getLong(importedRecordSetId), 0);
```

**Comment by DaniSzimko** [comment]
Thank you for your review @carolmariaabb, I committed the requested changes.

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:stop

**Comment by carolmariaabb** [comment]
@DaniSzimko I believe the following test failure is related to your changes, could you please take a look? Thanks!

1. [com.liferay.dynamic.data.lists.internal.exportimport.test.DDLDisplayExportImportTest](https://test-1-35.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/599423/testReport/com.liferay.dynamic.data.lists.internal.exportimport.test/DDLDisplayExportImportTest)

**Comment by DaniSzimko** [comment]
The problem was that I did not add the recordSetKey to the portletPreferences in testExportImport test case. I did a fixup, thank you for your review @carolmariaabb

**Comment by carolmariaabb** [comment]
@DaniSzimko please see https://github.com/liferay-forms/liferay-portal/pull/2302

**Pull request closed without merging**